praw/models/subreddits.py

```diff
@@ -85,28 +88,18 @@ def search_by_name(self, query, include_nsfw=True, exact=False):
         :param exact: Return only exact matches to ``query`` (default: False).
 
         """
-        result = self._reddit.post(
+        data = self._reddit.request(
+            "POST",
             API_PATH["subreddits_name_search"],
             data={
                 "include_over_18": include_nsfw,
                 "exact": exact,
                 "query": query,
             },
         )
-        return [self._reddit.subreddit(x) for x in result["names"]]
-
-    def search_by_topic(self, query):
```

Comment:
For future PRs it'd be great if separate things, like removals, were pulled out into, at the very least, their own commits, but separate PRs would be best.

Reply:
I’ve moved this change into a separate commit.